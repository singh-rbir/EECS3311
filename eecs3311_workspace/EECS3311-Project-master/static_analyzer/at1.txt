reset

add_division
type_check
evaluate

reset

add_division
add_integer_constant (5)
add_integer_constant (2)
type_check
evaluate 