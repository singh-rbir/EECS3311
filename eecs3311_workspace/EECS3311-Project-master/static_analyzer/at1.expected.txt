  Expression currently specified: ?
  Report: Expression is initialized.
->reset
  Expression currently specified: ?
  Report: Error (Initial expression cannot be reset).
->add_division
  Expression currently specified: (? / nil)
  Report: OK.
->type_check
  Expression currently specified: (? / nil)
  Report: Error (Expression is not yet fully specified).
->evaluate
  Expression currently specified: (? / nil)
  Report: Error (Expression is not yet fully specified).
->reset
  Expression currently specified: ?
  Report: OK.
->add_division
  Expression currently specified: (? / nil)
  Report: OK.
->add_integer_constant(5)
  Expression currently specified: (5 / ?)
  Report: OK.
->add_integer_constant(2)
  Expression currently specified: (5 / 2)
  Report: OK.
->type_check
  Expression currently specified: (5 / 2)
  Report: (5 / 2) is type-correct.
->evaluate
  Expression currently specified: (5 / 2)
  Report: 2
